openedx/features/wikimedia_features/admin_dashboard/static/admin_dashboard/js/course_reports.js

```diff
@@ -56,24 +87,128 @@
+            error: function(error) {
+                errorMessage = JSON.parse(error.responseText);
+                $('.request-response-error').text(errorMessage);
                 return $('.request-response-error').css({
                     display: 'block'
                 });
             },
             success: function(data) {
-                ReportDownloads();
                 $('.request-response').text(data.status);
-                return $('.msg-confirm').css({
-                    display: 'block'
-                });
             }
         });
     });
 
-}).call(this);
+    $("input[name='list-anon-ids']").click(function() {
+        let url_for_list_anon_ids = '/courses/' + course_name + '/instructor/api/get_anon_ids';
+        $.ajaxSetup({
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken')
+            },
+        });
+        $.ajax({
+            type: 'POST',
+            dataType: 'json',
+            url: url_for_list_anon_ids,
+            error: function(error) {
+                errorMessage = JSON.parse(error.responseText);
+                $('.request-response-error').text(errorMessage);
+            },
+            success: function(data) {
+                $('.request-response').text(data.status);
+            }
+        });
+    });
+
+    $("input[name='problem-grade-report']").click(function() {
+        let url_for_problem_grade_report = '/courses/' + course_name + '/instructor/api/problem_grade_report';
+        $.ajaxSetup({
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken')
+            },
+        });
+        console.log(url_for_problem_grade_report);
+        $.ajax({
+            type: 'POST',
+            dataType: 'json',
+            url: url_for_problem_grade_report,
+            error: function(error) {
+                errorMessage = JSON.parse(error.responseText);
+                $('.request-response-error').text(errorMessage);
+            },
+            success: function(data) {
+                $('.request-response').text(data.status);
+            }
+        });
+    });
+
+    $("input[name='list-may-enroll-csv']").click(function() {
+        let url_for_list_may_enroll = '/courses/' + course_name + '/instructor/api/get_students_who_may_enroll';
+        $.ajaxSetup({
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken')
+            },
+        });
+        console.log(url_for_list_may_enroll);
+        $.ajax({
+            type: 'POST',
+            dataType: 'json',
+            url: url_for_list_may_enroll,
+            error: function(error) {
+                errorMessage = JSON.parse(error.responseText);
+                $('.request-response-error').text(errorMessage);
+            },
+            success: function(data) {
+                $('.request-response').text(data.status);
+            }
+        });
+    });
+
+    $("input[name='average-calculate-grades-csv']").click(function() {
```

Comment:
can we write separate method for ajax call that receives url as parameter ? or maybe we can attach extra data to the button for the url path and use that.